typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -6,8 +8,29 @@ import { getCoreEnvironmentConfig, getEnvironment } from '../utils';
 
 import { getApp } from './utils';
 
+const constMetricLabels = {
+  // this needs to get set in main because of async reasons
+  abacus_deployment: '',
+  abacus_context: 'abacus',
+};
+
+const metricsRegister = new Registry();
+const messagesCount = new Counter({
+  name: 'abacus_kathy_messages',
+  help: 'Test messages which have been sent with',
+  registers: [metricsRegister],
+  labelNames: [
+    'origin',
+    'remote',
+    'status',
+    ...(Object.keys(constMetricLabels) as (keyof typeof constMetricLabels)[]),
+  ],
+});
+metricsRegister.registerMetric(messagesCount);
```

Comment:
1) Metrics really should be static because they are a global count and passing them around is annoying
2) `registerMetric` does not need to run after `abacus_deployment`, all this does in the static context is pull the label lame (hence `Object.keys` and not `Object.values`).